pkg/steps/artifacts.go

```diff
@@ -559,41 +563,52 @@ func containerHasVolumeName(container coreapi.Container, name string) bool {
 	return true
 }
 
-func addArtifactsToTemplate(template *templateapi.Template) {
-	for i := range template.Objects {
-		t := &template.Objects[i]
-		var pod map[string]interface{}
-		if err := json.Unmarshal(t.Raw, &pod); err != nil {
-			log.Printf("error: object can't be unmarshalled: %v", err)
-			continue
-		}
-		if jsonString(pod, "kind") != "Pod" || jsonString(pod, "apiVersion") != "v1" {
-			continue
-		}
-		if !arrayHasObjectString(jsonArray(pod, "spec", "volumes"), "name", "artifacts") {
-			continue
-		}
-		names := allPodContainerNamesWithArtifacts(pod)
-		if len(names) == 0 {
-			continue
-		}
-		data, err := json.Marshal(artifactsContainer())
-		if err != nil {
-			panic(err)
+func addArtifactsToTemplateObject(object *runtime.RawExtension) error {
```

Comment:
We should keep the old separation of logic -- process the template as one thing and add artifacts to all the pods it has inside of it that need them. This lets us test more of the logic and puts less of it into the mega-method. If you want a smaller chunk for unit tests, have a function `addArtifactsToTemplateObject` that takes a `Pod`